catalog/spatial/geocoding/spatial-geocoding-feature/src/test/java/org/codice/ddf/spatial/geocoding/feature/TestCatalogFeatureQueryable.java

```diff
@@ -0,0 +1,96 @@
+package org.codice.ddf.spatial.geocoding.feature;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.vividsolutions.jts.geom.Geometry;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.filter.proxy.builder.GeotoolsFilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.opengis.feature.simple.SimpleFeature;
+
+public class TestCatalogFeatureQueryable {
+  private FilterBuilder filterBuilder = new GeotoolsFilterBuilder();
+
+  private CatalogFramework catalogFramework;
+
+  private CatalogFeatureQueryable catalogFeatureQueryable;
+
+  static final String WKT_STRING = "POLYGON((30 10, 10 20, 20 40, 40 40, 30 10))";
+  static final String WKT_TYPE = "Polygon";
+  static final int WKT_NUM_POINTS = 5;
+
+  @Before
+  public void setUp() {
+    catalogFramework = mock(CatalogFramework.class);
+    catalogFeatureQueryable = new CatalogFeatureQueryable();
+    catalogFeatureQueryable.setCatalogFramework(catalogFramework);
+    catalogFeatureQueryable.setFilterBuilder(filterBuilder);
+  }
+
+  @Test
+  public void testQuery()
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+
+    final ArgumentCaptor<QueryRequest> requestArgument =
+        ArgumentCaptor.forClass(QueryRequest.class);
+
+    // stub query response (list of metacards), test how features are returned
+    QueryResponse queryResponse = getMockQueryResponse();
+    when(catalogFramework.query(any())).thenReturn(queryResponse);
+
+    List<SimpleFeature> results = catalogFeatureQueryable.query("JAM", 1);
+
+    verify(catalogFramework, atLeastOnce()).query(requestArgument.capture());
+
+    SimpleFeature result = results.get(0);
+    Geometry geometry = (Geometry) result.getDefaultGeometry();
+    assertThat(geometry.getNumPoints(), is(WKT_NUM_POINTS));
+    assertThat(geometry.getGeometryType(), is(WKT_TYPE));
+
+    QueryImpl query = (QueryImpl) requestArgument.getValue().getQuery();
+    assertThat(
+        query.getFilter().toString(),
```

Comment:
Is there a better way than `toString` to assert what the filter is?

Reply:
Didn't look like via getFilter() at least. I might have to go up a level and set expectations on the `FilterBuilder` instead? Was trying to avoid some of the testing bloat but agree this could be brittle.